README.md
Tux Disc
========

<center><img src="./tux-disc.png"
alt="Tux Disc" width="292" height="300" /></center>

---

[![](https://img.shields.io/badge/Mastodon-gray.svg?&style=for-the-badge&logo=mastodon)](https://niu.moe/@easthighNerd)
![](https://img.shields.io/badge/Matrix-@easthighnerd:ubports.chat-hotpink.svg?&style=for-the-badge&logo=matrix)

---

[![GitHub release](https://img.shields.io/github/release/easthighNerd/Tux-Disc.svg?&style=for-the-badge)](https://github.com/easthighNerd/Tux-Disc/releases/latest)
[![GitHub release date](https://img.shields.io/github/release-date/easthighNerd/Tux-Disc.svg?&style=for-the-badge)](https://github.com/easthighNerd/Tux-Disc/tree/master)
[![GitHub license](https://img.shields.io/github/license/easthighNerd/Tux-Disc.svg?&style=for-the-badge)](./LICENSE)
[![GitHub last commit](https://img.shields.io/github/last-commit/easthighNerd/Tux-Disc/version-2.svg?&style=for-the-badge)](https://github.com/easthighNerd/Tux-Disc/commits/master)
[![GitHub top language](https://img.shields.io/github/languages/top/easthighNerd/Tux-Disc.svg?&style=for-the-badge&logo=linux)](https://github.com/easthighNerd/Tux-Disc/tree/master)
=======
[![GitHub release date](https://img.shields.io/github/release-date/easthighNerd/Tux-Disc.svg?&style=for-the-badge)](https://github.com/easthighNerd/Tux-Disc/tree/master)
[![GitHub license](https://img.shields.io/github/license/easthighNerd/Tux-Disc.svg?&style=for-the-badge)](./LICENSE)
[![GitHub last commit](https://img.shields.io/github/last-commit/easthighNerd/Tux-Disc/version-2.svg?&style=for-the-badge)](https://github.com/easthighNerd/Tux-Disc/commits/master)
[![GitHub top language](https://img.shields.io/github/languages/top/easthighNerd/Tux-Disc.svg?&style=for-the-badge&logo=linux)](https://github.com/easthighNerd/Tux-Disc/tree/master)

---

[![](https://img.shields.io/badge/Changelog-blue.svg?&style=for-the-badge)](./CHANGELOG.md)[![](https://img.shields.io/badge/Distro%20List-brightgreen.svg?&style=for-the-badge)](./distro-list.md)

---

Tux Disc is a command line utility for downloading ISO images for Ubuntu and it's flavors, as well as KDE neon.

Tux Disc provides a simple, easy to use interface, that makes it a breeze to download Ubuntu ISO images.

This is pretty much a little project of mine born from my wanting to choose which Ubuntu flavor's 17.10 daily testing ISO I wanted to download, and just kinda ended up becoming this whole thing for downloading a bunch of Linux distributions install images.

Screenshots
===========
<center><img src="./screenshots/main-menu.png"
alt="Main Menu" /></center>
<center><img src="./screenshots/downloading.png"
alt="Downloading" /></center>

Get Tux Disc
============
You can get Tux Disc by compiling the source code.

To compile and install Tux Disc from source,
* Download the source code, by either
  * going to the [releases][2] page
    * download the source code either as .zip or .tar.gz
    * extract the contents of the archive
  * using `git` to clone the repository
  * click on `Clone or Download` and click on `Download ZIP`
    * extract the contents of the archive
* Open a terminal window and `cd` into the folder for Tux Disc
* Run
  ```
  # ./build.sh
  ```


Once completed, Tux Disc can now be run either from your desktop's application launcher, or by running
```
$ tux-disc
```

Planned Features
================

In no particular order, these are some features/milestones that I have planned for future development

* Graphical user interface (buttons everywhere!)
* Flatpak/Snap packaging (for easier cross-distro distribution)
* Complete recoding into Python (Might make the GUI easier, and should make it easier to port to other OSes)

[1]: ./CHANGELOG.md
[2]: https://github.com/easthighNerd/tux-disc/releases
[3]: ./distro-list.md


CHANGELOG.md
Changelog
=========

2.1-1
-----
### Added
* `Ubuntu 20.04`

### Changed
* Ubuntu `18.04.3` > `18.04.4`
* LTS versions of Ubuntu are now numbered based ona variable, making it a lot easier for me to update them as needed

### Removed
* `Ubuntu 19.04`

2.0-1
-----
### Added
* `16.04.6`, `18.04.3`, `19.04`, & `19.10` releases of all Ubuntu flavors

### Changed
* Rewrote script from scratch
* Simplified installation into a single `build.sh`
* Updated KDE neon versions and respective links
* distro-list.md now reflects the updated distros available for download
* Actual script no longer has version number in filename

### Removed
* All distros that are not (a flavor of) Ubuntu or KDE neon
* Ubuntu GNOME (rest in peace old friend)
* Outdated versions of Ubuntu and it's flavors

1.11-1
------

### Added
* Kubuntu 18.04 Beta 1 32/64-bit
* Ubuntu Budgie 18.04 Beta 1 32/64-bit
* Ubuntu MATE 18.04 Beta 1 32/64-bit
* Xubuntu 18.04 Beta 1 32/64-bit

1.10-2
------

### Added
* Versioning file to more easily change 16.04 point releases for each new point release

### Changed
* Qubes 4.0 Release Candidate 4 > 5
* Changed how Desktop file gets installed

1.10-1
------

### Added
* Kali Linux KDE 64-bit
* Kali Linux LXDE 64-bit
* Kali Linux MATE 64-bit
* Kali Linux Xfce 64-bit

### Changed
* Kali Linux (Light) 2017.3 > 2018.1
* Qubes 4.0 RC2 > RC4

1.10-1
------

### Added
* Kali Linux KDE 64-bit
* Kali Linux LXDE 64-bit
* Kali Linux MATE 64-bit
* Kali Linux Xfce 64-bit

### Changed
* Kali Linux (Light) 2017.3 > 2018.1
* Qubes 4.0 RC2 > RC4

1.9-1
-----
### Added
* Kali Linux 32/64-bit
* Kali Linux Light 32/64-bit

### Changed
* Kubuntu 17.10 > 17.10.1
* Lubuntu 17.10 > 17.10.1
* Ubuntu 17.10 > 17.10.1
* Ubuntu Budgie 17.10 > 17.10.1
* Ubuntu MATE 17.10 > 17.10.1
* Ubuntu Studio 17.10 > 17.10.1
* Xubuntu 17.10 > 17.10.1

### Removed
* Kubuntu 17.04
* Lubuntu 17.04
* Ubuntu 17.04
* Ubuntu Budgie 17.04
* Ubuntu GNOME 17.04
* Ubuntu MATE 17.04
* Ubuntu Studio 17.04
* Xubuntu 17.04

1.8-1
-----
### Changed
* Rework of entire script
* Script size reduced to 1/3 previous size

### Removed
* Ubuntu MATE PowerPC and ARMv7 temporarily removed due to rework

1.7-1 _(The lost version)_
-----
### New
* Proper package created and signed
* PPA available for easy software installation and updating
* ppa:flamestar/tux-disc

### Added
* Linux Mint 18.3
* - Cinnamon desktop 32bit and 64bit
* - MATE desktop 32bit and 64bit

2017-11-16-v1.2
---------------
### Added
* Option to download a specific branch _(source only)_

### Changed
* Removed leftover code no longer being used

2017-11-16-v1.1
---------------
### Fixed
* Ubuntu Studio downloads

2017-11-16-v1.0
---------------
### Added
* **Fedora Workstation 27**
* Net Install (i386/amd64)
* Live Install (amd64)

### Changed
* **Fedora 26**
* Now in separate section under "Fedora"
* Now labeled as "Workstation 26"

2017-11-12-v1.0
---------------
### Changed
* Renamed application from Linux ISO Downloader to Tux Disc

Older Versions
==============

2017-11-09-v1.0
---------------

### Added
* Kubuntu 18.04 Daily (i386/amd64)
* Lubuntu 18.04 Daily (i386/amd64)
* Ubuntu Budgie 18.04 Daily (i386/amd64)
* Ubuntu MATE 18.04 Daily (i386/amd64)
* Xubuntu 18.04 Daily (i386/amd64)

2017-11-08-v1.1
---------------

### Added
* Stylization to beginning of script

2017-11-08-v1.0
---------------

### Changed
* Version no longer includes hour and minute of build

### Added
* Ubuntu 18.04 Daily (amd64)

2017-10-30_20-36_v1.0
---------------------

### Updated
* Updated Debian Binary to new version

### Added
* Fedora 26 Net Install (i386/amd64)
* Fedora 26 Live Install (i386/amd64)
* Qubes 3.2 (amd64)
* Qubes 4.0-rc2 (amd64)

2017-10-29_20-26_v1.3
---------------------

### Updated
* Updated to Revision v1.3

### Added
* Added build scripts to compile from source
* New icon featuring Tux holding a CD (in reference to the downloading of Linux ISO images)
* Icon added to application launcher for easy access (icon stored in ~/.local/share/applications/)

### Changed
* Slimmed down binary

2017-10-29_20-26_v1.1
---------------------

### Changed
* ARMv7 to armhf
* PowerPC to powerpc

2017-10-29_20-26_v1.0
---------------------

### Added
* Ubuntu Studio 16.04 LTS (i386/amd64)
* Ubuntu Studio 17.04 (i386/amd64)
* Ubuntu Studio 17.10 (i386/amd64)

2017-10-19_05-53
----------------

### Added
* Kubuntu 17.10 (i386/amd64)
* Lubuntu 17.10 (i386/amd64)
* Ubuntu 17.10 (amd64)
* Ubuntu Budgie 17.10 (i386/amd64)
* Ubuntu MATE 17.10 (i386/amd64)
* Xubuntu 17.10 (i386/amd64)

### Removed
* Kubuntu 17.10 Daily (i386/amd64)
* Lubuntu 17.10 Daily (i386/amd64)
* Ubuntu 17.10 Daily (amd64)
* Ubuntu Budgie 17.10 Daily (i386/amd64)
* Ubuntu MATE 17.10 Daily (i386/amd64)
* Xubuntu 17.10 Daily (i386/amd64)


tux-disc_1.10-1.sh
#!/bin/bash

# Establishes base directory of script
BASEDIR=$(dirname $0)

# Prints todays date (YYYYmmdd) when called by $DATE
DATE=$(date '+%Y%m%d')

# Establishes binary version number
BINARYVER=$(grep "" $BASEDIR/binaryver)

# Establishes patch version
PATCHVER=$(grep "" $BASEDIR/patchver)

# Establishes Debian version
DEBVER=$(grep "" $BASEDIR/debver)

# Establishes Linux Mint version
MINTVER=$(grep "" $BASEDIR/mintver)

# Establishes Kali Linux version
KALIVER=$(grep "" $BASEDIR/kaliver)

echo "
                        ##############################
                        ##                          ##
                        ##           Tux            ##
                        ##                          ##
                        ##           Disc           ##
                        ##                          ##
                        ##      Version $BINARYVER-$PATCHVER      ##
                        ##                          ##
                        ##############################
"

# Linux distrobution
echo 'Please select from the following'
select DISTRO in "Debian $DEBVER" "Fedora" "Kali Linux $KALIVER" "KDE Neon" "Kubuntu" "Linux Mint $MINTVER" "Lubuntu" "Qubes" "Ubuntu" "Ubuntu Budgie" "Ubuntu GNOME" "Ubuntu MATE" "Ubuntu Studio" "Xubuntu"
do
        case $DISTRO in
        "Debian $DEBVER"|"Fedora"|"Kali Linux $KALIVER"|"KDE Neon"|"Kubuntu"|"Linux Mint $MINTVER"|"Lubuntu"|"Qubes"|"Ubuntu"|"Ubuntu Budgie"|"Ubuntu GNOME"|"Ubuntu MATE"|"Ubuntu Studio"|"Xubuntu")
                break
                ;;
        *)
                echo "Please select a distrobution"
                ;;
        esac
done

# Debian
if [[ $DISTRO = "Debian $DEBVER" ]]; then
# Install method
    echo "Now select an install method for $DISTRO"
    select RELEASE in "Net Install" "Full Install" "Live Install"
    do
            case $RELEASE in
            "Net Install"|"Full Install"|"Live Install")
                    break
                    ;;
            *)
                    echo "Please select an install method"
                    ;;
            esac
    done
# Net Install
    if [[ $RELEASE = "Net Install" ]]; then
# Architecture
        echo "Now select an architecture for $DISTRO $RELEASE"
        select ARC in "32-bit" "64-bit"
        do
                case $ARC in
                "32-bit"|"64-bit")
                        break
                        ;;
                *)
                        echo "Please select an architecture"
                        ;;
                esac
        done

        if [[ $ARC = "32-bit" ]]; then
            arc=$"i386"
        fi

        if [[ $ARC = "64-bit" ]]; then
            arc=$"amd64"
        fi

        FINAL=$"$DISTRO $ARC $RELEASE"

# Download
            OPTION=$"-P" URL=$" https://cdimage.debian.org/debian-cd/current/$arc/iso-cd/debian-$DEBVER-$arc-netinst.iso"
    fi
# Full Install
    if [[ $RELEASE = "Full Install" ]]; then
# Architecture
        echo "Now select an architecture for $DISTRO $RELEASE"
        select ARC in "32-bit" "64-bit"
        do
                case $ARC in
                "32-bit"|"64-bit")
                        break
                        ;;
                *)
                        echo "Please select an architecture"
                        ;;
                esac
        done

        if [[ $ARC = "32-bit" ]]; then
            arc=$"i386"
        fi

        if [[ $ARC = "64-bit" ]]; then
            arc=$"amd64"
        fi

        FINAL=$"$DISTRO $ARC $RELEASE"
# Download
            OPTION=$"-P"
            URL1=$" https://cdimage.debian.org/debian-cd/current/$arc/iso-dvd/debian-$DEBVER-$arc-DVD-1.iso"
            URL2=$" https://cdimage.debian.org/debian-cd/current/$arc/iso-dvd/debian-$DEBVER-$arc-DVD-2.iso"
            URL3=$" https://cdimage.debian.org/debian-cd/current/$arc/iso-dvd/debian-$DEBVER-$arc-DVD-3.iso"
    fi
# Live Install
    if [[ $RELEASE = "Live Install" ]]; then
# Architecture
        echo "Now select an architecture for $DISTRO $RELEASE"
        select ARC in "32-bit" "64-bit"
        do
                case $ARC in
                "32-bit"|"64-bit")
                        break
                        ;;
                *)
                        echo "Please select an architecture"
                        ;;
                esac
        done
# Which desktop?
            echo "Which desktop would you like for $DISTRO $ARC $RELEASE?"
            select DEBDESK in Cinnamon GNOME KDE LXDE MATE XFCE
            do
                    case $DEBDESK in
                    Cinnamon|GNOME|KDE|LXDE|MATE|XFCE)
                            break
                            ;;
                    *)
                            echo "Please select a desktop"
                            ;;
                    esac
            done

            FINAL=$"$DISTRO $RELEASE $ARC $DEBDESK"

            debdesk="${DEBDESK,,}"

            if [[ $ARC = "32-bit" ]]; then
                arc=$"i386"
            fi

            if [[ $ARC = "64-bit" ]]; then
                arc=$"amd64"
            fi
# Download
                OPTION=$"-P" URL=$" https://cdimage.debian.org/debian-cd/current-live/$arc/iso-hybrid/debian-live-$DEBVER-$arc-$debdesk.iso"
            fi
fi
# Fedora
if [[ $DISTRO = "Fedora" ]]; then
  # Fedora version
      echo "Please select your version of $DISTRO for download"
      select FED in "Workstation 26" "Workstation 27"
      do
              case $FED in
              "Workstation 26"|"Workstation 27")
                      break
                      ;;
              *)
                      echo "Please select a Fedora version"
                      ;;
              esac
      done
# Install method
    echo "Now select an install method for $DISTRO $FED"
    select RELEASE in "Net Install" "Live Install"
    do
            case $RELEASE in
            "Net Install"|"Live Install")
                    break
                    ;;
            *)
                    echo "Please select an install method"
                    ;;
            esac
    done
# Net install 26
    if [[ $FED = "Workstation 26" ]] && [[ $RELEASE = "Net Install" ]]; then
# Architecture
        echo "Now select an architecture for $DISTRO $FED $RELEASE"
        select ARC in "32-bit" "64-bit"
        do
                case $ARC in
                "32-bit"|"64-bit")
                        break
                        ;;
                *)
                        echo "Please select an architecture"
                        ;;
                esac
        done
# i386
        if [[ $ARC = "32-bit" ]]; then
# Download
            OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora-secondary/releases/26/Workstation/i386/iso/Fedora-Workstation-netinst-i386-26-1.5.iso"
        fi
# amd64
        if [[ $ARC = "64-bit" ]]; then
# Download
            OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora/linux/releases/26/Workstation/x86_64/iso/Fedora-Workstation-netinst-x86_64-26-1.5.iso"
        fi
    fi
# Live Install 26
    if [[ $FED = "Workstation 26" ]] && [[ $RELEASE = "Live Install" ]]; then
# Architecture
        echo "Now select an architecture for $DISTRO $FED $RELEASE"
        select ARC in "32-bit" "64-bit"
        do
                case $ARC in
                "32-bit"|"64-bit")
                        break
                        ;;
                *)
                        echo "Please select an architecture"
                        ;;
                esac
        done
# i386
        if [[ $ARC = "32-bit" ]]; then
# Download
            OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora-secondary/releases/26/Workstation/i386/iso/Fedora-Workstation-Live-i386-26-1.5.iso"
        fi
# amd64
        if [[ $ARC = "64-bit" ]]; then
# Download
            OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora/linux/releases/26/Workstation/x86_64/iso/Fedora-Workstation-Live-x86_64-26-1.5.iso"
        fi
    fi

    # Net install 27
        if [[ $FED = "Workstation 27" ]] && [[ $RELEASE = "Net Install" ]]; then
    # Architecture
            echo "Now select an architecture for $DISTRO $FED $RELEASE"
            select ARC in "32-bit" "64-bit"
            do
                    case $ARC in
                    "32-bit"|"64-bit")
                            break
                            ;;
                    *)
                            echo "Please select an architecture"
                            ;;
                    esac
            done
    # i386
            if [[ $ARC = "32-bit" ]]; then
    # Download
                OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora-secondary/releases/27/Workstation/i386/iso/Fedora-Workstation-netinst-i386-27-1.6.iso"
            fi
    # amd64
            if [[ $ARC = "64-bit" ]]; then
    # Download
                OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora/linux/releases/27/Workstation/x86_64/iso/Fedora-Workstation-netinst-x86_64-27-1.6.iso"
            fi
        fi
    # Live Install 27
        if [[ $FED = "Workstation 27" ]] && [[ $RELEASE = "Live Install" ]]; then

                ARC=$"64-bit"
    # Download
                OPTION=$"-P" URL=$" https://download.fedoraproject.org/pub/fedora/linux/releases/27/Workstation/x86_64/iso/Fedora-Workstation-Live-x86_64-27-1.6.iso"
        fi
    FINAL=$"$DISTRO $FED $ARC $RELEASE"
fi
# Kali Linux
if [[ $DISTRO = "Kali Linux $KALIVER" ]]; then
  echo "Now select a release for $DISTRO"
  select RELEASE in "Kali Linux" "Kali Linux KDE" "Kali Linux Light" "Kali Linux LXDE" "Kali Linux MATE" "Kali Linux Xfce"
  do
          case $RELEASE in
          "Kali Linux"|"Kali Linux KDE"|"Kali Linux Light"|"Kali Linux LXDE"|"Kali Linux MATE"|"Kali Linux Xfce")
                  break
                  ;;
          *)
                  echo "Please select a release"
                  ;;
          esac
  done
  if [[ $RELEASE = "Kali Linux" ]] || [[ $RELEASE = "Kali Linux Light" ]]; then
    echo "Now select an architecture for $RELEASE $KALIVER"
    select ARC in "32-bit" "64-bit"
    do
            case $ARC in
            "32-bit"|"64-bit")
                    break
                    ;;
            *)
                    ;;
            esac
    done
  fi

  if [[ ! $ARC = "32-bit" ]] && [[ ! $ARC = "64-bit" ]]; then
    ARC=$"64-bit"
  fi

  if [[ $ARC = "32-bit" ]]; then
    arc=$"i386"
  fi

  if [[ $ARC = "64-bit" ]]; then
    arc=$"amd64"
  fi

  HYPHEN="${RELEASE// /-}"

  distro="${HYPHEN,,}"

  OPTION=$"-P"

  URL=$" http://cdimage.kali.org/kali-$KALIVER/$distro-$KALIVER-$arc.iso"

  FINAL=$"$RELEASE $KALIVER $ARC"
fi
# KDE Neon
if [[ $DISTRO = "KDE Neon" ]]; then
# Release
    echo "Now select a release for $DISTRO"
    select RELEASE in "User Edition" "User LTS Edition" "Developer Edition Git-Unstable" "Developer Edition Git-Stable"
    do
            case $RELEASE in
            "User Edition"|"User LTS Edition"|"Developer Edition Git-Unstable"|"Developer Edition Git-Stable")
                    break
                    ;;
            *)
                    echo "Please select a release"
                    ;;
            esac
    done
# User Edition
    if [[ $RELEASE = "User Edition" ]]; then
# Download
            OPTION=$"-O" URL=$"/neon-useredition-current-amd64.iso https://files.kde.org/neon/images/neon-useredition/current/neon-useredition-current.iso"
    fi
# User LTS Edition
    if [[ $RELEASE = "User LTS Edition" ]]; then
# Download
            OPTION=$"-O" URL=$"/neon-userltsedition-current-amd64.iso https://files.kde.org/neon/images/neon-userltsedition/current/neon-userltsedition-current.iso"
    fi
# Developer Edition Git-Unstable
    if [[ $RELEASE = "Developer Edition Git-Unstable" ]]; then
# Download
            OPTION=$"-O" URL=$"/neon-devedition-gitunstable-current-amd64.iso https://files.kde.org/neon/images/neon-devedition-gitunstable/current/neon-devedition-gitunstable-current.iso"
    fi
# Developer Edition Git-Stable
    if [[ $RELEASE = "Developer Edition Git-Stable" ]]; then
# Download
            OPTION=$"-O" URL=$"/neon-devedition-gitstable-current-amd64.iso https://files.kde.org/neon/images/neon-devedition-gitstable/current/neon-devedition-gitstable-current.iso"
    fi

    FINAL=$"$DISTRO $RELEASE 64-bit"
fi
# Linux Mint
if [[ $DISTRO = "Linux Mint $MINTVER" ]]; then
    echo "Which desktop would you like?"
    select MINTDESK in Cinnamon KDE MATE Xfce
    do
            case $MINTDESK in
            Cinnamon|KDE|MATE|Xfce)
                    break
                    ;;
            *)
                    echo "Please select a desktop"
                    ;;
            esac
    done

    echo "What architecture will you be running?"
    select ARC in 32-bit 64-bit
    do
            case $ARC in
            32-bit|64-bit)
                    break
                    ;;
            *)
                    echo "Please select an architecture"
                    ;;
            esac
    done

    arc="${ARC//-/}"

    mintdesk="${MINTDESK,,}"

    FINAL=$"$DISTRO $MINTDESK $ARC"

    OPTION=$"-P" URL=$" http://mirrors.evowise.com/linuxmint/stable/$MINTVER/linuxmint-$MINTVER-$mintdesk-$arc.iso"
fi
# Qubes
if [[ $DISTRO = "Qubes" ]]; then
# Release
    echo "Now select a release for $DISTRO"
    select RELEASE in "3.2" "4.0-rc4"
    do
            case $RELEASE in
            "3.2"|"4.0-rc4")
                    break
                    ;;
            *)
                    echo "Please select a release"
                    ;;
            esac
    done
# Qubes 3.2
    if [[ $RELEASE = "3.2" ]]; then
# Message
            FINAL=$"$DISTRO $RELEASE 64-bit"
# Download
            OPTION=$"-P" URL=$" https://mirrors.kernel.org/qubes/iso/Qubes-R3.2-x86_64.iso"
    fi
# Qubes 4.0-rc2
    if [[ $RELEASE = "4.0-rc4" ]]; then
# Message
            FINAL=$"$DISTRO $RELEASE 64-bit"
# Download
            OPTION=$"-P" URL=$" https://mirrors.kernel.org/qubes/iso/Qubes-R4.0-rc4-x86_64.iso"
    fi
fi
# Ubuntu
if [[ "$DISTRO" = "Kubuntu" ]] || [[ "$DISTRO" = "Lubuntu" ]] || [[ "$DISTRO" = "Ubuntu" ]] || [[ "$DISTRO" = "Ubuntu Budgie" ]] || [[ $DISTRO = "Ubuntu GNOME" ]] || [[ "$DISTRO" = "Ubuntu MATE" ]] || [[ "$DISTRO" = "Ubuntu Studio" ]] || [[ "$DISTRO" = "Xubuntu" ]]; then

    if [[ $DISTRO = "Ubuntu Studio" ]]; then
        HYPHEN="${DISTRO// /}"
        DESKDVD=$"dvd"
    else
        HYPHEN="${DISTRO// /-}"
        DESKDVD=$"desktop"
    fi

    distro="${HYPHEN,,}"

    if [[ $DISTRO = "Ubuntu Budgie" ]]; then
        echo 'Please select a release'
        select RELEASE in "17.10.1" "18.04 Daily"
        do
                case $RELEASE in
                "17.10.1"|"18.04 Daily")
                        break
                        ;;
            *)
                        echo 'Please select a release'
                        ;;
                esac
        done
    fi

    if [[ $DISTRO = "Ubuntu GNOME" ]]; then
        echo 'Please select a release'
        select RELEASE in "16.04.3"
        do
                case $RELEASE in
                "16.04.3")
                        break
                        ;;
                *)
                        echo 'Please select a release'
                        ;;
                esac
        done
    fi

    if [[ $DISTRO = "Ubuntu Studio" ]]; then
        echo 'Please select a release'
        select RELEASE in "16.04.3" "17.10.1"
        do
                case $RELEASE in
                "16.04.3"|"17.10.1")
                        break
                        ;;
                *)
                        echo 'Please select a release'
                        ;;
                esac
        done
    fi

    if [[ "$DISTRO" = "Kubuntu" ]] || [[ "$DISTRO" = "Lubuntu" ]] || [[ "$DISTRO" = "Ubuntu" ]] || [[ "$DISTRO" = "Ubuntu MATE" ]] || [[ "$DISTRO" = "Xubuntu" ]]; then
        echo 'Please select a release'
        select RELEASE in "16.04.3" "17.10.1" "18.04 Daily"
        do
                case $RELEASE in
                "16.04.3"|"17.10.1"|"18.04 Daily")
                        break
                        ;;
                *)
                        echo 'Please select a release'
                        ;;
                esac
        done
    fi

    if ( [[ $DISTRO = "Ubuntu" ]] && [[ $RELEASE = "17.10.1" ]] ) || ( [[ $DISTRO = "Ubuntu" ]] && [[ $RELEASE = "18.04 Daily" ]] ); then
        ARC=$"64-bit"
    else
        echo 'Please select an architecture'
        select ARC in "32-bit" "64-bit"
        do
                case $ARC in
                "32-bit"|"64-bit")
                        break
                        ;;
                *)
                        echo 'Please select an architecture'
                        ;;
                esac
        done
    fi

    if [[ $ARC = "32-bit" ]]; then
        arc=$"i386"
    fi

    if [[ $ARC = "64-bit" ]]; then
        arc=$"amd64"
    fi

    FINAL=$"$DISTRO $RELEASE $ARC"

    if [[ $DISTRO = "Ubuntu" ]]; then
        if [[ $RELEASE = "18.04 Daily" ]]; then
            OPTION=$"-O" URL=$"/$distro-bionic-$DESKDVD-$DATE-$arc.iso http://cdimage.ubuntu.com/$distro/daily-live/current/bionic-$DESKDVD-$arc.iso"
        else
            OPTION=$"-P" URL=$" http://releases.ubuntu.com/$RELEASE/$distro-$RELEASE-$DESKDVD-$arc.iso"
        fi
    else
        if [[ $RELEASE = "18.04 Daily" ]]; then
             OPTION=$"-O" URL=$"/$distro-bionic-$DESKDVD-$DATE-$arc.iso http://cdimage.ubuntu.com/$distro/daily-live/current/bionic-$DESKDVD-$arc.iso"
        else
            OPTION=$"-P" URL=$" http://cdimage.ubuntu.com/$distro/releases/$RELEASE/release/$distro-$RELEASE-$DESKDVD-$arc.iso"
        fi
    fi
  fi

  read -p "Is this what you wanted to download? $FINAL (Y/n) " YN
  case "$YN" in
    y|Y ) YN=$"yes";;
    n|N ) YN=$"no";;
   *) YN=$"invalid" && echo "invalid";;
  esac

if [[ $YN = "yes" ]]; then

  read -p "Where in your home directory would you like to download $FINAL? (d for default) " DIRECT
  case "$DIRECT" in
    d|D ) DIRECT=$"ISO";;
   *);;
  esac

  echo "Now downloading $FINAL to $HOME/$DIRECT..."

  if [ ! -d "$HOME/$DIRECT" ]; then
      mkdir $HOME/$DIRECT
  fi

  if [[ $DISTRO = "Debian $DEBVER" ]] && [[ $RELEASE = "Full Install" ]]; then
    wget $OPTION $HOME/$DIRECT$URL1
    wget $OPTION $HOME/$DIRECT$URL2
    wget $OPTION $HOME/$DIRECT$URL3
  else
    wget $OPTION $HOME/$DIRECT$URL
  fi

fi

  if [[ $YN = "no" ]] || [[ $YN = "invalid" ]]; then
    bash $BASEDIR/tux-disc_$BINARYVER-$PATCHVER.sh
  fi

  read -p "Would you like to download another ISO? (Y/n) " REPEAT
  case "$REPEAT" in
    y|Y ) REPEAT=$"yes";;
    n|N ) REPEAT=$"no";;
   *) REPEAT=$"invalid" && echo "invalid";;
  esac

# Repeat
if [[ $REPEAT = "yes" ]]; then
# Message
    echo "Here we go again"
# Change directory
    cd $BASEDIR
# Relaunch script
    bash tux-disc_$BINARYVER-$PATCHVER.sh
fi
# End
if [[ $REPEAT = "no" ]] || [[ $REPEAT = "invalid" ]]; then
# Message
    echo "Goodbye"
fi
